CRM/Xdedupe/Resolver/CustomGroupIntegral.php

```diff
@@ -0,0 +1,185 @@
+<?php
+/*-------------------------------------------------------+
+| SYSTOPIA's Extended Deduper                            |
+| Copyright (C) 2025 SYSTOPIA                            |
+| Author: B. Endres (endres@systopia.de)                 |
+|         S. Frank (frank@systopia.de)                   |
+| http://www.systopia.de/                                |
++--------------------------------------------------------+
+| This program is released as free software under the    |
+| Affero GPL license. You can redistribute it and/or     |
+| modify it under the terms of this license which you    |
+| can read by viewing the included agpl.txt or online    |
+| at www.gnu.org/licenses/agpl.html. Removal of this     |
+| copyright header is strictly prohibited without        |
+| written permission from the original author(s).        |
++--------------------------------------------------------*/
+
+declare(strict_types = 1);
+use CRM_Xdedupe_ExtensionUtil as E;
+
+/**
+ * Implements a resolver for contact custom group conflicts.
+ *
+ * This version will treat the custom group data
+ * in an integral way, i.e. the fields of the group should be treated
+ * as one record set
+ */
+class CRM_Xdedupe_Resolver_CustomGroupIntegral extends CRM_Xdedupe_Resolver {
+
+  /**
+   * @var integer ID of the custom group
+   */
+  protected int $custom_group_id;
+
+  protected ?array $custom_group_data = null;
+
+  /**
+   * @param CRM_Xdedupe_Merge $merge
+   * @param mixed $custom_group_id
+   */
+  public function __construct($merge, $custom_group_id) {
+    $this->custom_group_id = (int) $custom_group_id;
```

Comment:
If `$custom_group_id` can be anything, shouldn't it be validated before casting it to int?